dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java

```diff
@@ -59,60 +68,102 @@
+                                new Thread(() -> System.exit(e.getErrorCode()), "Checkpoint exit.").start();
+                            }
+                            // TODO should we always exit on failure?
+                        }
+                    }
+                }
+
+                @Override
+                public void failedCompletion(Future<Boolean> future, Throwable t) {
+                }
+            });
+        }
+    }
+    
+    @Override
+    public void check() {
+        if (checkpointApplications) {
+            try {
+                checkpoint(Phase.APPLICATIONS);
+            } catch (CheckpointFailed e) {
+                if (e.getType() == Type.SNAPSHOT_FAILED) {
+                    new Thread(() -> System.exit(e.getErrorCode()), "Checkpoint exit.").start();
+                }
+                // TODO should we always exit on failure?
+            }
+        }
     }
 
     /**
-     * Perform a snapshot for the specified phase. The result of the
-     * snapshot will be stored in the specified directory.
-     * Before the snapshot is taken the registered {@link CheckpointHookFactory#create(Phase)}
+     * Perform a checkpoint for the specified phase. The result of the
+     * checkpoint will be stored in the specified directory.
+     * Before the checkpoint is done the registered {@link CheckpointHookFactory#create(Phase)}
      * methods are called to obtain the {@link CheckpointHook} instances which will participate
-     * in the prepare and restore steps for the snapshot process.
-     * Each snapshot hook instance will their {@link CheckpointHook#prepare(Phase)}
-     * methods are called before the snapshot is taken. After the snapshot
-     * is taken each snapshot hook instance will have their {@link CheckpointHook#restore(Phase)}
+     * in the prepare and restore steps for the checkpoint process.
+     * Each checkpoint hook instance will their {@link CheckpointHook#prepare(Phase)}
+     * methods are called before the checkpoint is done. After the checkpoint
+     * is done each checkpoint hook instance will have their {@link CheckpointHook#restore(Phase)}
      * methods called.
      *
      * @deprecated Provided to support debugging from osgi console.
-     * @param phase the phase to take the snapshot
-     * @throws SnapshotFailed if the snapshot fails
+     * @param phase the phase to take the checkpoint
+     * @throws CheckpointFailed if the checkpoint fails
      */
     @Deprecated
-    @Descriptor("Take a snapshot")
-    public void snapshot(@Parameter(names = "-p", absentValue = "") @Descriptor("The phase to snapshot") Phase phase,
-                         @Descriptor("Directory to store the snapshot") File directory) throws SnapshotFailed {
-        doSnapshot(phase, directory);
+    @Descriptor("Do a checkpoint")
+    public void checkpoint(@Parameter(names = "-p", absentValue = "") @Descriptor("The phase to checkpoint") Phase phase,
```

Comment:
We should get rid of this gogo command method altogether.  I don't think we have any need for it anymore.  The method is still used by unit tests so we can make it package private.